src/memory_targets/memory_target_numa.c

```diff
@@ -87,22 +87,34 @@ static umf_result_t numa_memory_provider_create_from_memspace(
             return UMF_RESULT_ERROR_INVALID_ARGUMENT;
         }
     } else {
-        params.numa_mode = UMF_NUMA_MODE_BIND;
+        if (memspace == umfMemspaceHostAllGet()) {
```

Comment:
This is bad . What you are doing here is a pointer comparison.  While what you want to do is check if memspace contains all numanodes in system. For now it is very difficult to get all numanodes in system from different source then memspaceHostAllGet() - but really soon it will be possible. 

What you should have here is a helper function which checks if memspace contains all nodes, or function to compare namespaces.

Reply:
ok yes, this is a ptr comparison, but the ‘memspace’ I'm comparing here is the handle returned from umfMemspaceHostAllGet(), and this function always returns the same handle for each call. Furthermore, comparing the contents of memspace is not the same as comparing memspace itself - and here I want to check exactly whether the user is using the default (host all) memspace